src/components/Dashboard/SafeApps/Grid.tsx

```diff
@@ -0,0 +1,112 @@
+import { ReactElement, useMemo } from 'react'
+import styled from 'styled-components'
+import { Button } from '@gnosis.pm/safe-react-components'
+import { generatePath, Link } from 'react-router-dom'
+
+import { useAppList } from 'src/routes/safe/components/Apps/hooks/appList/useAppList'
+import { GENERIC_APPS_ROUTE } from 'src/routes/routes'
+import Card from 'src/components/Dashboard/SafeApps/Card'
+import ExploreIcon from 'src/assets/icons/explore.svg'
+import local from 'src/utils/storage/local'
+import { SafeApp } from 'src/routes/safe/components/Apps/types'
+import { APPS_DASHBOARD, AppTrackData, rankTrackedSafeApps } from 'src/routes/safe/components/Apps/trackAppUsageCount'
+import Skeleton from '@material-ui/lab/Skeleton/Skeleton'
+
+const StyledGrid = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 20px;
+  flex-wrap: wrap;
+`
+
+const SkeletonWrapper = styled.div`
+  border-radius: 8px;
+  overflow: hidden;
+`
+
+const StyledExplorerButton = styled.div`
+  width: 260px;
+  height: 200px;
+  background-color: white;
+  border-radius: 8px;
+  padding: 24px;
+
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  gap: 8px;
+`
+
+const StyledLink = styled(Link)`
+  text-decoration: none;
+
+  > button {
+    width: 200px;
+  }
+`
+
+// Transactions Builder && Wallet connect
+const officialAppIds = ['29', '11']
+
+const Grid = ({ size = 3 }: { size?: number }): ReactElement => {
+  const { allApps, pinnedSafeApps, togglePin, isLoading } = useAppList()
+
+  const displayedApps = useMemo(() => {
+    const trackData = local.getItem<AppTrackData>(APPS_DASHBOARD) || {}
```

Comment:
I'd suggest extracting this into a function so that all loading/saving functionality is in one place. You have to reference the key every time otherwise.